lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/HttpTrackingURLProvider.java

```diff
@@ -36,37 +36,79 @@
    */
   private static final Logger LOG = Logger.getLogger(HttpTrackingURLProvider.class.getName());
 
+  /** Default hostname to provide in the Application Master registration. */
+  private static final String AM_REGISTRATION_HOST = "";
+
+  /** Default port number to provide in the Application Master registration. */
+  private static final int AM_REGISTRATION_PORT = -1;
+
   /**
    * HttpServer.
    */
   private final HttpServer httpServer;
 
   /**
+   * Driver host.
+   */
+  private final String host;
+
+  /**
+   * port number of the htt server.
+   */
+  private final int port;
+
+  /**
+   * Tracking Url of the ttp server.
+   */
+  private final String trackingUrl;
+
+  /**
    * Constructor of HttpTrackingURLProvider.
    *
    * @param httpServer
    */
   @Inject
   public HttpTrackingURLProvider(final HttpServer httpServer) {
     this.httpServer = httpServer instanceof NoOpHttpServerImpl ? null : httpServer;
+
+    try {
+      host = InetAddress.getLocalHost().getHostAddress();
```

Comment:
By the way, why do we always get the IP address? Shouldn't we return the host name instead? AFAIR host name is preferred (if not required) for YARN.